app/src/main/java/ch/epfl/swissteam/services/ExtendedChatRelation.java

```diff
@@ -28,11 +41,34 @@ public ChatRelation getChatRelation_() {
         return chatRelation_;
     }
 
+    public long getTimestamp_() { return timestamp_; }
+
     @Override
     public boolean equals(Object o){
         if(o == null || o.getClass() != ExtendedChatRelation.class) {return false;}
         else{
             return chatRelation_.equals(((ExtendedChatRelation)o).getChatRelation_());
         }
     }
+
+    public void retrieveLastTimestamp(DBCallBack<Long> callBackTimestamp){
+        Query lastQuery = DBUtility.get().getDb_()
+                .child(DBUtility.CHATS).child(chatRelation_.getId_()).orderByKey().limitToLast(1);
+        lastQuery.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                Iterator<DataSnapshot> iter = dataSnapshot.getChildren().iterator();
+                ChatMessage cM = null;
```

Comment:
```suggestion
                ChatMessage chatMessage = null;
```

Reply:
No point in shortening a variable name if it could lead to misunderstandings.

Reply:
No point in not shortening a variable if it is just to check non-null and that this variable exist for 5 lines